pkg/kubelet/config/apiserver.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Reads the pod configuration from the Kubernetes apiserver.
+package config
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+	"github.com/golang/glog"
+)
+
+type podLW interface {
+	List() (runtime.Object, error)
+	Watch(resourceVersion string) (watch.Interface, error)
+}
+
+type realPodLW struct {
+	client   *client.Client
+	hostname string
+}
+
+func (lw realPodLW) List() (runtime.Object, error) {
+	// TODO: add a field selector to pkg/client.PodInterface so that can be used.
+	return lw.client.
+		Get().
+		Namespace(api.NamespaceAll).
+		Resource("pods").
+		SelectorParam("fields", labels.OneTermEqualSelector("Status.Host", lw.hostname)).
+		Do().
+		Get()
+}
+
+func (lw realPodLW) Watch(resourceVersion string) (watch.Interface, error) {
+	// TODO: add a field selector to pkg/client.PodInterface so that can be used.
+	return lw.client.
+		Get().
+		Prefix("watch").
+		Namespace(api.NamespaceAll).
+		Resource("pods").
+		SelectorParam("fields", labels.OneTermEqualSelector("Status.Host", lw.hostname)).
+		Param("resourceVersion", resourceVersion).
+		Watch()
+}
+
+var _ cache.ListerWatcher = realPodLW{}
+
+// NewSourceApiserver creates a config source that watches and pulls from the apiserver.
+func NewSourceApiserver(client *client.Client, hostname string, updates chan<- interface{}) {
+	lw := realPodLW{
+		client:   client,
+		hostname: hostname,
+	}
+	newSourceApiserverFromLW(lw, updates)
+}
+
+// newSourceApiserverFromLW holds creates a config source that watches an pulls from the apiserver.
+func newSourceApiserverFromLW(lw podLW, updates chan<- interface{}) {
+	send := func(objs []interface{}) {
+		var bpods []api.BoundPod
+		for _, o := range objs {
+			pod := o.(*api.Pod)
```

Comment:
BoundPodFactory can't be easily reused.  And I plan to delete it once this series of changes finishes.

I reviewed what MakeBoundPod does:
  - `api.Scheme.Convert` -- doing that already
  - set env vars -- I intend this to be done in a different part of kubelet in a different PR
  - sets selflink -- Need to add this.